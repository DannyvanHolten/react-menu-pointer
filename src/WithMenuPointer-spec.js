import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import withMenuPointer from './WithMenuPointer';

describe('WithExperiments', () => {
  it('should render correctly', () => {
    const Component = withMenuPointer(<p>Component</p>);
    const wrapper = shallow(<Component />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should handle slope correctly', () => {
    const Component = withMenuPointer(<p>Component</p>);
    const wrapper = shallow(<Component />);
    expect(wrapper.instance().slope({ x: 3, y: 5 }, { x: 10, y: 267 })).toEqual(37.42857142857143);
  });

  it('should handle getDelay correctly', () => {
    const Component = withMenuPointer(<p>Component</p>);
    const wrapper = shallow(<Component />);
    expect(wrapper.instance().slope({ x: 3, y: 5 }, { x: 10, y: 267 })).toEqual(37.42857142857143);
  });
});
