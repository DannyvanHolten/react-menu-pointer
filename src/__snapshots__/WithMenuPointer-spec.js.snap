// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithExperiments should render correctly 1`] = `
<[object Object]
  config={Object {}}
  menuPointer={
    Object {
      "handleMenuLeave": [Function],
      "handleRowEnter": [Function],
    }
  }
/>
`;
